### ISSUE TITLE
Handle ores on different stone types.

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#611]
There are three options to handle:
1. don't, let the addon creator handle it(→ lots of duplicate work, but would allow making special textures for each ore and stone combination)
2. Generate a bunch of new block types during startup for all combinations of ores and stone. This would cost a bunch of "index space" reducing the maximum number of possible blocks.
3. Use rotation mode and store the source block in the 16 bit data section.

And the textures could either be auto-generated and stored in a texture, or recalculated in a shader.

**Comment by careeoki** [comment]
Third option. 
A transparent ore texture could be overlayed onto the base stone texture.

**Comment by IntegratedQuantum** [comment]
> And the textures could either be auto-generated and stored in a texture, or recalculated in a shader.

The best option would be to just draw the overlay in a separate face, just like it's done for the block breaking animation.
This avoids the explosion of resources from storing generated textures, and it also means the shader has to be less complex.

**Comment by careeoki** [comment]
Yeah, that's what I mean, just like how the breaking animation is overlaid.
